app/src/main/java/com/github/fribourgsdp/radio/game/GameActivity.kt

```diff
@@ -306,6 +322,29 @@ open class GameActivity : AppCompatActivity(), GameView, Timer.Listener {
             else null
     }
 
+
+    override fun addHint(songNameHint: SongNameHint) {
+        if(withHint) {
+            hintTextView.visibility = View.VISIBLE
+            this.songNameHint = songNameHint
+            lastTime = 0
+            hintTextView.text = this.songNameHint.toString()
+        }
+    }
+
+    fun updateHint(timeInSeconds: Int){
+        if(withHint && hintTextView.visibility == View.VISIBLE) {
+            val limit = ceil(gameDuration/ (ceil(songNameHint.length()/2.0)+1))
+            if(timeInSeconds - lastTime >= limit) {
+                songNameHint.addALetter()
+                hintTextView.text = this.songNameHint.toString()
+                lastTime = timeInSeconds
+            }
+        }
+    }
```

Comment:
This algorithm makes it such that at the end of the timer, half of the letters in the song's title have been displayed. Is this maybe too much? I think we should ask the group to see if 1/3 of the total title is enough.

Reply:
I think half the letters are good as they appear one after the other but if others think like you I will change.

Reply:
We tested (with @nathanduchesne) on scribble and at the end there were only 2 letters on small words. I think 1/2 is a bit too much, as one can guess the title just by building the words from the letters that have appeared. 1/3 or 1/4 letters seems a betters choice. Changing the value shouldn't be too difficult if we have to change it anyway.

Reply:
Thanks for changing it to 1/3. Everything is fine for me now.